diff --git a/Source/Core/Element.cpp b/Source/Core/Element.cpp
--- a/Source/Core/Element.cpp
+++ b/Source/Core/Element.cpp
@@ -1565,6 +1565,12 @@ void Element::OnPropertyChange(const PropertyNameList& changed_properties)
 		}
 	}
 
+	if (changed_properties.find(OPACITY) != changed_properties.end())
+	{
+		// Reload decorators so that the opacity value can be applied by decorators supporting it
+		GetElementDecoration()->ReloadDecorators();
+	}
+
 	// Dirty the background if it's changed.
 	if (all_dirty ||
 		changed_properties.find(BACKGROUND_COLOR) != changed_properties.end() ||
